lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h

```diff
@@ -423,19 +428,34 @@ class SwiftASTContext : public TypeSystemSwift {
 
   CompilerType GetErrorType() override;
 
-  bool HasErrors();
+  /// Error handling
+  /// \{
+  bool HasErrors() const;
+  bool HasClangImporterErrors() const;
+
+  void RaiseDiagnostic(DiagnosticSeverity severity, llvm::StringRef message);
+  void RaiseFatalError(std::string msg) { m_fatal_errors.SetErrorString(msg); }
+  static bool HasFatalErrors(swift::ASTContext *ast_context);
+  bool HasFatalErrors() const {
+    return m_fatal_errors.Fail() || HasFatalErrors(m_ast_context_ap.get());
+  }
+
+  Status GetAllErrors() const;
```

Comment:
Can this return an `llvm::Error` (that must be checked) instead?

Reply:
GetAllErrors() doesn't consume the errors, so I don't think that's needed.